templates/ovndbcluster/bin/setup.sh

```diff
@@ -73,4 +79,44 @@ set "$@" --ovn-${DB_TYPE}-log=-vconsole:{{ .OVN_LOG_LEVEL }}
 set "$@" --ovn-${DB_TYPE}-logfile=/dev/null
 
 # don't log to file (we already log to console)
-$@ ${OPTS} run_${DB_TYPE}_ovsdb -- -vfile:off
+$@ ${OPTS} run_${DB_TYPE}_ovsdb -- -vfile:off &
+
+# Once the database is running, we will attempt to configure db options
+CTLCMD="ovn-${DB_TYPE}ctl --no-leader-only"
+
+# TODO: this logic is dedicated for the leader, not necessarily the -0 pod. The
+# assumption that -0 is always the leader is invalid and a bug that should be fixed.
+if [[ "$(hostname)" == "{{ .SERVICE_NAME }}-0" ]]; then
+    # The command will wait until the daemon is connected and the DB is available
+    # All following ctl invocation will use the local DB replica in the daemon
+    export OVN_${DB_TYPE^^}_DAEMON=$(${CTLCMD} --pidfile --detach)
+
+{{- if .TLS }}
+    ${CTLCMD} set-ssl {{.OVNDB_KEY_PATH}} {{.OVNDB_CERT_PATH}} {{.OVNDB_CACERT_PATH}}
+    ${CTLCMD} set-connection ${DB_SCHEME}:${DB_PORT}:0.0.0.0
```

Comment:
if there is revision can also fix 0.0.0.0 -> ${DB_ADDR}

Reply:
Done. Btw I was wondering if we need that at all, since we already do

```
set "$@" --db-${DB_TYPE}-addr=${DB_ADDR}
set "$@" --db-${DB_TYPE}-port=${DB_PORT}
{{- if .TLS }}
set "$@" --ovn-${DB_TYPE}-db-ssl-key={{.OVNDB_KEY_PATH}}
set "$@" --ovn-${DB_TYPE}-db-ssl-cert={{.OVNDB_CERT_PATH}}
set "$@" --ovn-${DB_TYPE}-db-ssl-ca-cert={{.OVNDB_CACERT_PATH}}
```

above.

But decided not to touch it for now.